### PR TITLE
Downloader: params_ is now private

### PR DESCRIPTION
**Pull request by mtrojan-ub**

### CODE REVIEW

File: cpp/lib/src/Downloader.cc [hunk 1]
```diff
@@ -706,6 +679,38 @@ bool Downloader::getHttpEquivRedirect(std::string * const redirect_url) const {
...
+
+void Downloader::setProxy(const std::string &proxy_host_and_port) {
+    params_.proxy_host_and_port_ = proxy_host_and_port;
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_USERAGENT, proxy_host_and_port.c_str()) != CURLE_OK))
+        throw std::runtime_error("Downloader::setProxy(): curl_easy_setopt() failed! (" + proxy_host_and_port + ")");
+}
+
+
+void Downloader::setUserAgent(const std::string &user_agent) {
+    params_.user_agent_ = user_agent;
```
Comment (ruschein) [inline cpp/lib/src/Downloader.cc]: Do we need to consider what we want to happen when "user_agent" is empty or don't we?

File: cpp/lib/include/Downloader.h [hunk 2]
```diff
@@ -167,6 +167,12 @@ class Downloader {
     /** \note Get HTTP response code */
     unsigned getResponseCode();
 
+    void setAcceptableLanguages(const std::string &acceptable_languages);
+    void setHonourRobotsDotTxt(const bool honour_robots_dot_txt) { params_.honour_robots_dot_txt_ = honour_robots_dot_txt; };
+    void setIgnoreSslCertificates(const bool ignore_ssl_certificates);
+    void setProxy(const std::string &proxy_host_and_port);
+    void setUserAgent(const std::string &user_agent);
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/lib/include/Downloader.h]: added comment

File: cpp/lib/src/Downloader.cc [hunk 3]
```diff
@@ -335,137 +333,112 @@ void Downloader::init() {
...
+        throw std::runtime_error("in Downloader::init: curl_easy_setopt() failed (10)!");
 
     // User agent information:
-    if (user_agent->empty())
-        *user_agent = Downloader::DEFAULT_USER_AGENT_STRING;
-    if (unlikely(::curl_easy_setopt(*easy_handle, CURLOPT_USERAGENT, user_agent->c_str()) != CURLE_OK))
-        throw std::runtime_error("in Downloader::InitCurlEasyHandle: curl_easy_setopt() failed (10)!");
+    if (params_.user_agent_.empty())
+        params_.user_agent_ = Downloader::DEFAULT_USER_AGENT_STRING;
+    setUserAgent(params_.user_agent_);
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/lib/src/Downloader.cc]: Logic was moved inside setUserAgent

File: cpp/lib/src/Downloader.cc [hunk 4]
```diff
@@ -706,6 +679,38 @@ bool Downloader::getHttpEquivRedirect(std::string * const redirect_url) const {
...
+    params_.ignore_ssl_certificates_ = ignore_ssl_certificates;
+    long curl_value(ignore_ssl_certificates ? 0L : 1L);
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_SSL_VERIFYPEER, curl_value) != CURLE_OK))
+        throw std::runtime_error("in Downloader::setIgnoreSslCertificates(): curl_easy_setopt() failed!");
+    if (unlikely(::curl_easy_setopt(easy_handle_, CURLOPT_SSL_VERIFYHOST, curl_value) != CURLE_OK))
+        throw std::runtime_error("in Downloader::setIgnoreSslCertificates(): curl_easy_setopt() failed!");
+}
+
+
+void Downloader::setProxy(const std::string &proxy_host_and_port) {
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/lib/src/Downloader.cc]: Yes, by passing empty string, comment in Downloader.h added

File: cpp/lib/src/Downloader.cc [hunk 5]
```diff
@@ -62,7 +61,7 @@ static curl_slist *RemoveAllWithMatchingPrefix(curl_slist * const slist, const c
     while (current != nullptr) {
         curl_slist *next(current->next);
 
-        if (std::strncmp(current->data, prefix, prefix_length) != 0)
+        if (strncasecmp(current->data, prefix, prefix_length) != 0)
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/lib/src/Downloader.cc]: done

File: cpp/lib/src/Downloader.cc (hunk 1 above)
Comment (mtrojan-ub) [reply cpp/lib/src/Downloader.cc]: Yes, see comment in Downloader.h

**Pull request merged**